### PR TITLE
KYLIN-4793 In some Hadoop versions, kylin calcite reported error beca…

### PR DESCRIPTION
**Pull request by zhangayqian**
…use there is no guava14

## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Kylin?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have create an issue on [Kylin's jira](https://issues.apache.org/jira/browse/KYLIN), and have described the bug/feature there in detail
- [x] Commit messages in my PR start with the related jira ID, like "KYLIN-0000 Make Kylin project open-source"
- [x] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If this change need a document change, I will prepare another pr against the `document` branch
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at user@kylin or dev@kylin by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zhangayqian** [comment]
Commit https://github.com/apache/kylin/commit/e1fd472d00eea438d26c26f032e819db56e33486 remove guava14 from Kylin.war，which will causes calcite reported error because there is no guava14 in some Hadoop versions.

File: build/bin/kylin.sh [hunk 1]
```diff
@@ -72,7 +72,14 @@ function retrieveDependency() {
     KYLIN_SPARK_JARS_HOME="${KYLIN_HOME}/spark/jars"
     if [[ -d "/usr/hdp/current/hdinsight-zookeeper" && $hdp_version == "2"* ]]
     then
-       echo "The current Hadoop environment is HDI3, will replace some jars package for ${KYLIN_HOME}/spark/jars"
+       echo "The current Hadoop environment is HDI3, will remove ${KYLIN_HOME}/tomcat/webapps/kylin/WEB-INF/lib/guava-14.0.jar and replace some jars package for ${KYLIN_HOME}/spark/jars"
+       unzip ${KYLIN_HOME}/tomcat/webapps/kylin.war
```
Comment (zzcclp) [inline build/bin/kylin.sh:76]: It's better to check whether exists unzip command, if not, echo some error message and exit.

File: build/bin/kylin.sh [hunk 2]
```diff
@@ -72,7 +72,14 @@ function retrieveDependency() {
...
+       if [[ -f ${KYLIN_HOME}/tomcat/webapps/kylin/WEB-INF/lib/guava-14.0.jar ]]
+       then
+          rm -rf ${KYLIN_HOME}/tomcat/webapps/kylin/WEB-INF/lib/guava-14.0.jar
+          jar -cvfM0 ${KYLIN_HOME}/tomcat/webapps/kylin.war ${KYLIN_HOME}/tomcat/webapps/kylin/
```
Comment (zzcclp) [inline build/bin/kylin.sh:80]: IMO, don't need to re-war this package.

File: build/bin/kylin.sh (hunk 1 above)
Comment (zhangayqian) [reply build/bin/kylin.sh:76]: My negligence. I wiil use `jar -xvf` to decompression kylin.war.

File: build/bin/kylin.sh (hunk 2 above)
Comment (zhangayqian) [reply build/bin/kylin.sh]: I will delete kylin.war after decompression.

**Comment by zzcclp** [comment]
LGTM, @hit-lacus please review.

**Review by hit-lacus — APPROVED** [review]
Approved

**Pull request merged**